core/src/main/java/io/syndesis/core/Tokens.java

```diff
@@ -119,24 +136,11 @@ public static String fetchProviderTokenFromKeycloak(TokenProvider provider) {
             );
         }
 
-        switch (provider) {
-            case OPENSHIFT:
-                ObjectMapper om = new ObjectMapper();
-                try {
-                    AccessTokenResponse accessToken = om.readValue(responseBody, AccessTokenResponse.class);
-                    return accessToken.getToken();
-                } catch (IOException e) {
-                    throw SyndesisServerException.launderThrowable(e);
-                }
-            case GITHUB:
-                Optional<NameValuePair> accessToken = URLEncodedUtils.parse(responseBody, StandardCharsets.UTF_8)
-                    .stream()
-                    .filter((nameValuePair) -> nameValuePair.getName().equalsIgnoreCase("access_token")).findFirst();
-                return accessToken.orElseThrow(IllegalStateException::new).getValue();
-            default:
-                throw new IllegalArgumentException("Unknown provider type: " + provider);
+        if (provider != TokenProvider.OPENSHIFT && provider != TokenProvider.GITHUB) {
```

Comment:
tbh I would even ommit this check and let the enum itself throw the IllegalArgumentException in apply() if it is not supported.